drivers/python/rethinkdb/net.py

```diff
@@ -462,16 +458,18 @@ def _read_response(self, token, deadline=None):
                 self._parent.reconnect(noreply_wait=False)
                 raise ex
 
-            # Construct response
-            res = Response(res_token, res_buf)
+            res = None
 
-            cursor = self._cursor_cache.get(res.token)
+            cursor = self._cursor_cache.get(res_token)
             if cursor is not None:
```

Comment:
This code needs to handle the case where the result is not for a cursor (e.g. `r.expr(1)`).